src/document.rs

```diff
@@ -314,14 +328,18 @@ impl Document {
                 .unwrap_or(default_excerpt_separator);
 
             let excerpt = if let Some(excerpt_str) = excerpt_attr {
-                excerpt_str
+                excerpt_str.to_string()
             } else if excerpt_separator.is_empty() {
-                ""
+                String::new()
```

Comment:
`"".to_string()` also allocates a new string :)

What error do you get if you do `try!(self.render_html(..., context, source))` in every clause instead?